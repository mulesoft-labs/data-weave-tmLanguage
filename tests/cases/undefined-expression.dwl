fun func5 <T>(identifier: String): T = ???
var x: Boolean = ???
var z: true = ???
var y: {x?: Null} = ???
type NotDefined = ???
{
 "lala" : ???
}