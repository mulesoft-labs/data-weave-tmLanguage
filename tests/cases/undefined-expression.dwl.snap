>fun func5 <T>(identifier: String): T = ???
#^^^ source.data-weave meta.directive.fun.dw storage.type.dw
#   ^ source.data-weave meta.directive.fun.dw
#    ^^^^^ source.data-weave meta.directive.fun.dw entity.name.function.dw
#         ^ source.data-weave meta.directive.fun.dw
#          ^ source.data-weave meta.directive.fun.dw
#           ^ source.data-weave meta.directive.fun.dw entity.name.type.parameter.dw
#            ^ source.data-weave meta.directive.fun.dw
#             ^ source.data-weave meta.directive.fun.dw
#              ^^^^^^^^^^ source.data-weave meta.directive.fun.dw variable.parameter.dw
#                        ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                         ^ source.data-weave meta.directive.fun.dw
#                          ^^^^^^ source.data-weave meta.directive.fun.dw support.class.dw
#                                ^ source.data-weave meta.directive.fun.dw
#                                 ^ source.data-weave meta.directive.fun.dw keyword.operator.declaration.dw
#                                  ^ source.data-weave meta.directive.fun.dw
#                                   ^ source.data-weave meta.directive.fun.dw support.class.dw
#                                    ^ source.data-weave meta.directive.fun.dw
#                                     ^ source.data-weave meta.directive.fun.dw keyword.operator.assignment.dw
#                                      ^ source.data-weave
#                                       ^^^ source.data-weave constant.language.undefined.dw
>var x: Boolean = ???
#^^^ source.data-weave meta.directive.var.dw storage.type.dw
#   ^ source.data-weave meta.directive.var.dw
#    ^ source.data-weave meta.directive.var.dw entity.name.variable.dw
#     ^ source.data-weave meta.directive.var.dw keyword.operator.declaration.dw
#      ^ source.data-weave meta.directive.var.dw
#       ^^^^^^^ source.data-weave meta.directive.var.dw support.class.dw
#              ^ source.data-weave meta.directive.var.dw
#               ^ source.data-weave meta.directive.var.dw keyword.operator.assignment.dw
#                ^ source.data-weave
#                 ^^^ source.data-weave constant.language.undefined.dw
>var z: true = ???
#^^^ source.data-weave meta.directive.var.dw storage.type.dw
#   ^ source.data-weave meta.directive.var.dw
#    ^ source.data-weave meta.directive.var.dw entity.name.variable.dw
#     ^ source.data-weave meta.directive.var.dw keyword.operator.declaration.dw
#      ^^^^^^ source.data-weave meta.directive.var.dw
#            ^ source.data-weave meta.directive.var.dw keyword.operator.assignment.dw
#             ^ source.data-weave
#              ^^^ source.data-weave constant.language.undefined.dw
>var y: {x?: Null} = ???
#^^^ source.data-weave meta.directive.var.dw storage.type.dw
#   ^ source.data-weave meta.directive.var.dw
#    ^ source.data-weave meta.directive.var.dw entity.name.variable.dw
#     ^ source.data-weave meta.directive.var.dw keyword.operator.declaration.dw
#      ^ source.data-weave meta.directive.var.dw
#       ^ source.data-weave meta.directive.var.dw
#        ^ source.data-weave meta.directive.var.dw variable.object.member.dw
#         ^ source.data-weave meta.directive.var.dw keyword.operator.optional.dw
#          ^ source.data-weave meta.directive.var.dw keyword.operator.declaration.dw
#           ^ source.data-weave meta.directive.var.dw
#            ^^^^ source.data-weave meta.directive.var.dw support.class.dw
#                ^ source.data-weave meta.directive.var.dw
#                 ^ source.data-weave meta.directive.var.dw
#                  ^ source.data-weave meta.directive.var.dw keyword.operator.assignment.dw
#                   ^ source.data-weave
#                    ^^^ source.data-weave constant.language.undefined.dw
>type NotDefined = ???
#^^^^ source.data-weave meta.directive.type.dw storage.type.dw
#    ^ source.data-weave meta.directive.type.dw
#     ^^^^^^^^^^ source.data-weave meta.directive.type.dw entity.name.type.dw
#               ^ source.data-weave meta.directive.type.dw
#                ^ source.data-weave meta.directive.type.dw keyword.other.dw
#                 ^ source.data-weave meta.directive.type.dw
#                  ^^^ source.data-weave meta.directive.type.dw constant.language.undefined.dw
>{
#^ source.data-weave meta.directive.type.dw
> "lala" : ???
#^ source.data-weave meta.directive.type.dw
# ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.begin.dw
#  ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#   ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#    ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#     ^ source.data-weave meta.directive.type.dw string.quoted.double.dw
#      ^ source.data-weave meta.directive.type.dw string.quoted.double.dw punctuation.definition.string.end.dw
#       ^ source.data-weave meta.directive.type.dw
#        ^ source.data-weave meta.directive.type.dw keyword.operator.declaration.dw
#         ^ source.data-weave meta.directive.type.dw
#          ^^^ source.data-weave meta.directive.type.dw constant.language.undefined.dw
>}
#^ source.data-weave meta.directive.type.dw